account_avatax_exemption/controller/main.py

```diff
@@ -0,0 +1,166 @@
+from odoo.tools import exception_to_unicode
+
+from odoo.addons.portal.controllers.portal import CustomerPortal
+
+_logger = logging.getLogger(__name__)
+
+
+class Exemption(http.Controller):
+    @http.route("/exemption/<int:exemption_id>", website=True, auth="public")
+    def get_exemption(self, **kw):
+        exemption_id = kw.get("exemption_id")
+        try:
+            message = (
+                request.env["res.partner.exemption"]
+                .sudo()
+                .search_exemption_line(exemption_id)
+            )
+        except Exception as e:
+            message = False, exception_to_unicode(e)
+        return request.render(
+            "account_avatax_exemption.exemption_page", {"message": message}
+        )
+
+
+class WebsiteExemption(CustomerPortal):
+    def _exemptions_domain(self, search=""):
+        """Get user's exemptions domain."""
+
+        avalara_salestax = (
+            request.env["avalara.salestax"]
+            .sudo()
+            .search([("exemption_export", "=", True)], limit=1)
+        )
+        domain = [("partner_id", "child_of", request.env.user.partner_id.id)]
+        if avalara_salestax.use_commercial_entity:
+            domain = [
+                (
+                    "partner_id",
+                    "child_of",
+                    request.env.user.partner_id.commercial_partner_id.id,
+                )
+            ]
+
+        return domain
+
+    def _prepare_portal_layout_values(self, exemption=None):
+        values = super()._prepare_portal_layout_values()
+        partner_counts = request.env["res.partner.exemption"].search_count(
+            self._exemptions_domain()
+        )
+        values["exemption_count"] = partner_counts
+        return values
+
+    def _prepare_exemptions_values(
+        self, page=1, date_begin=None, date_end=None, search="", sortby=None
+    ):
+        """Prepare the rendering context for the exemptions list."""
+        values = self._prepare_portal_layout_values()
+        Exemption = request.env["res.partner.exemption"]
+        base_url = "/my/exemptions"
+
+        searchbar_sortings = {
+            "date": {"label": _("Newest"), "order": "create_date desc"},
+            "expiry_date": {"label": _("Expiry Date"), "order": "expiry_date desc"},
+        }
+        if not sortby:
+            sortby = "date"
+        order = searchbar_sortings[sortby]["order"]
+
+        # Get the required domains
+        domain = self._exemptions_domain(search)
+
+        if date_begin and date_end:
+            domain += [
+                ("create_date", ">=", date_begin),
+                ("create_date", "<", date_end),
+            ]
+
+        # Make pager
+        pager = request.website.pager(
+            url=base_url,
+            url_args={"date_begin": date_begin, "date_end": date_end, "sortby": sortby},
+            total=Exemption.search_count(domain),
+            page=page,
+            step=self._items_per_page,
```

Comment:
```suggestion
            step=self._items_per_page,
```

After this commit https://github.com/odoo/odoo/pull/74996/commits/8a5e336ad31  the number of items per page has increased from `20` to `80`. We should consider adjusting this number to reduce the pressure on the database and improve the UI. Because if 80 items on 1 page is probably too much information.

Reply:
what is the convention to update? set self._items_per_page = 20 ?

Reply:
just a consideration. declare a new variable for example because in reality a lot of pagination has changed this value to 8, 12, 20,.. like the following example: https://github.com/odoo/odoo/blob/f60188fc9d2efc9e961fb4fecad800e529bee041/addons/website_hr_recruitment/controllers/main.py#L18

or we can keep the default `30` ([see](https://github.com/odoo/odoo/blob/f60188fc9d2efc9e961fb4fecad800e529bee041/addons/portal/controllers/portal.py#L17)) available in the function instead of overriding that value